File: Python/409. Longest Palindrome.py
class Solution:
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        maple = dict()
        for c in s:
            maple[c] = maple.get(c, 0) + 1
        ans = 0
        for k in maple:
            ans += (maple[k] // 2) * 2
        if ans != len(s):
            ans += 1
        return ans
                

File: C++/409. Longest Palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        int ans = 0;
        unordered_map<char, int> maple;
        for(auto c: s)
            maple[c] += 1;
        for(auto& c: maple)
            ans += (c.second/2)*2;
        return ans == s.length() ? ans : ans+1;
    }
};